contracts/upgradable/FinalProxy.sol

```diff
@@ -4,6 +4,7 @@ pragma solidity ^0.8.0;
 
 import { IProxy } from '../interfaces/IProxy.sol';
 import { IFinalProxy } from '../interfaces/IFinalProxy.sol';
+import { IUpgradable } from '../interfaces/IUpgradable.sol';
```

Comment:
Can we split contractId() into a separate interface (maybe `IContractIdentifier` or so)? Atm, we're pulling in the full IUpgradable which can be an issue for the bytecode if upgradable contract is changed

Reply:
The `IUpgradable`  interface most likely will stay the same event we we modify the `Upgradable`. Plus with Create3 we should not worried about the bytecode anymore.

Reply:
Added `IContractIdentifier` and made other contracts depend on it